Add Scone-OSV-Scan markdown file

laerson opened the pull request:
This pull request introduces a comprehensive deployment guide for the Confidential SCONE OSV Scan on Kubernetes by adding a new `deploy-sos.md` documentation file. The document provides step-by-step instructions, including environment validation, dependency checks, Helm setup, CAS attestation, MaxScale configuration, image building, and deployment automation. The guide is structured as a series of Bash functions to streamline and allow a working script to be extracted.

laerson committed:
docs(deploy-sos): add deployment guide for SCONE OSV Scan in K8s

scontain commented on deploy-sos.md:
```diff
@@ -0,0 +1,342 @@
+  sleep 5
+
+  scone cas attest localhost \
+    --mrsigner 195e5a6df987d6a515dd083750c1ea352283f8364d3ec9142b0d593988c6ed2d \
```
Please check with out two signers: standard and scone.cloud CAS

laerson committed:
docs(deploy-sos): use 6.0.x CLI image and accept both CAS signers

Address review feedback on PR #32. Switch SCONE_CLI_IMAGE from the old
sconecuratedimages/sconecli:alpine-scone5.9.0 to the canonical
registry.scontain.com/scone.cloud/sconecli:6.0.7 (stable per stable.txt).
Attest the CAS against the standard Scontain signer and fall back to the
scone.cloud signer, since scone cas attest --mrsigner overrides the single
default signer and is not repeatable.

Refs: #32
